File: seed/php-sdk/trace/src/V2/V3/Problem/Types/GeneratedFiles.php
<?php

namespace Seed\V2\V3\Problem\Types;

use Seed\Core\SerializableType;
use Seed\Commons\Types\Language;
use Seed\Core\JsonProperty;
use Seed\Core\ArrayType;

class GeneratedFiles extends SerializableType
{
    /**
     * @var array<value-of<Language>, Files> $generatedTestCaseFiles
     */
    #[JsonProperty('generatedTestCaseFiles'), ArrayType(["string" => Files::class])]
    public array $generatedTestCaseFiles;

    /**
     * @var array<value-of<Language>, Files> $generatedTemplateFiles
     */
    #[JsonProperty('generatedTemplateFiles'), ArrayType(["string" => Files::class])]
    public array $generatedTemplateFiles;

    /**
     * @var array<value-of<Language>, Files> $other
     */
    #[JsonProperty('other'), ArrayType(["string" => Files::class])]
    public array $other;

    /**
     * @param array{
     *   generatedTestCaseFiles: array<value-of<Language>, Files>,
     *   generatedTemplateFiles: array<value-of<Language>, Files>,
     *   other: array<value-of<Language>, Files>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->generatedTestCaseFiles = $values['generatedTestCaseFiles'];
        $this->generatedTemplateFiles = $values['generatedTemplateFiles'];
        $this->other = $values['other'];
    }
}

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/v_3/problem/generated_files.rb

module Seed
    module Types
        class GeneratedFiles < Internal::Types::Model
            field :generated_test_case_files, Internal::Types::Hash[Seed::commons::Language, Seed::v_2::v_3::problem::Files], optional: false, nullable: false
            field :generated_template_files, Internal::Types::Hash[Seed::commons::Language, Seed::v_2::v_3::problem::Files], optional: false, nullable: false
            field :other, Internal::Types::Hash[Seed::commons::Language, Seed::v_2::v_3::problem::Files], optional: false, nullable: false

    end
end
